### PR TITLE
`CircleCI`: use Xcode 14.2.0

### PR DESCRIPTION
**Pull request by NachoSoto**
See https://discuss.circleci.com/t/xcode-14-2-released-breaking-changes/46303

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -794,7 +794,7 @@ workflows:
         equal: [ scheduled_pipeline, << pipeline.trigger_source >> ]
     jobs:
       - prepare-next-version:
-          xcode_version: '14.1.0'
+          xcode_version: '14.2.0'
```
Comment (tonidero) [inline .circleci/config.yml:797]: Hmm I think we could probably run some of these jobs in non-mac machines... But that doesn't belong to this PR.

**Review by tonidero — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (NachoSoto) [reply .circleci/config.yml:797]: Yeah there's lots we could do to speed up some of these. Which ones were you thinking specifically?

Comment (NachoSoto) [reply .circleci/config.yml:797]: Oh like `prepare-next-version`, yeah. I think it's so we can have some basic stuff installed, but I'm sure they're lighter images we could use.

Comment (tonidero) [reply .circleci/config.yml:797]: Yeah, I was thinking on any like this, where it's just running some fastlane scripts.
